### ISSUE TITLE
Does max_seq_length specify the maxium number of words

### ISSUE THREAD

**Issue by artemlos** [huggingface/pytorch-pretrained-BERT#108]
I'm trying to figure out how the `--max_seq_length` parameter works in [run_classifier](https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/examples/run_classifier.py). Based on the source, it seems like it represents the number of words? Is that correct?

**Comment by rodgzilla** [comment]
`max_seq_length` specifies the maximum number of tokens of the input. The number of token is superior or equal to the number of words of an input. 

For example, the following sentence:

```
The man hits the saxophone and demonstrates how to properly use the racquet.
```

is tokenized as follows:

```
the man hits the saxophone and demonstrates how to properly use the ra ##c ##quet .
```

And depending on the task 2 to 3 additional special tokens (`[CLS]` and `[SEP]`) are added to the input to format it.

**Comment by artemlos** [comment]
@rodgzilla thanks!

**Comment by tsungruihon** [comment]
could we make it smaller?

**Comment by echan00** [comment]
So what if there are sentences where the maximum number of tokens is greater than max_seq_length?

Does that mean extra tokens beyond max_seq_length will get cut off?

**Comment by thomwolf** [comment]
@tsungruihon yes, just use smaller sentences

@echan00 no automatic cut off but there is a warning from the tokenizer that your inputs are too long and the model will throw an error. You have to limit the size manually.

**Comment by SaurabhBhatia0211** [comment]
Hi All,

Does that mean we cannot use BERT for classifying long documents. The documents having 5-6 Paragraphs and each paragraph having 10-15 mins with about 10-12 words in each line ?

**Comment by bavaria95** [comment]
@SaurabhBhatia0211 
You can try splitting a document to smaller chunks (e.g. paragraphs or even lines), computing embedding for each of those chunks, and average those vectors to get the document representation.

**Comment by brando90** [comment]
@rodgzilla is this true?

> HuggingFace's Trainer API, including the SFTrainer, by default pads all sequences to the maximum length within the batch, not to the max_seq_length argument. The max_seq_length argument serves as a hard limit to the sequence length, truncating any examples that are longer than that. The API was designed this way because padding to the maximum sequence length in the batch improves computational efficiency.

?
